From jochen@scram.de  Thu Jan 29 17:37:02 2004
X-Original-To: hmh@khazad-dum.debian.net
From: Jochen Friedrich <jochen@scram.de>
X-X-Sender: jochen@localhost
To: "Michael R. Schwarzbach" <mail@michael.schwarzbach.name>,
	230077@bugs.debian.org
Cc: Henrique de Moraes Holschuh <hmh@debian.org>
Subject: Re: Bug#230077: cyrus21-imapd: IPv6 support broken
Mime-Version: 1.0
Content-Transfer-Encoding: 8bit
Content-Type: text/plain; charset=iso-8859-1
Status: RO

Hi Michael,

> If you enable IPv6 in /etc/cyrus.conf (set proto to tcp6) native IPv6 is working,
> but the deamon is not listening on IPv4 anymore. It doesn't matter wether you
> setup the local 6to4 gateway (sit0). If you try to setup tcp and tcp6 in
> cyrus.conf the deamon doesn't start anymore with the error: imap used twice.
> It's not possible to enable IPv6 and IPv4 support simultaniously.

The bug is in the man-page not reflecting the IPv6 changes. The meaning of
proto is as following:

tcp4: only listen on IPv4
tcp6: only listen on IPv6
tcp:  listen on both IPv4 and IPv6

The same is valid for udp4, udp6 and udp, as well.

This patch should fix the man-page. It should be forwarded to upstream, as
well as cyrus 2.2 still doesn't list the proto arguments correctly
although the IPv6 patch is included.

Thanks,
--jochen

--- cyrus21-imapd-2.1.16.old/man/cyrus.conf.5	2004-01-29 20:26:16.000000000 +0100
+++ cyrus21-imapd-2.1.16/man/cyrus.conf.5	2004-01-29 20:32:55.000000000 +0100
@@ -110,8 +110,17 @@
 interface, and \fIport\fR is either a port number or service name (as listed
 in \fB/etc/services\fR).
 .IP "\fBproto=\fRtcp" 5
-The protocol used for this service (\fItcp\fR, \fIudp\fR).  This
-string argument is optional.
+The protocol used for this service (\fItcp\fR, \fItcp4\fR, \fItcp6\fR,
+\fIudp\fR, \fIudp4\fR, \fIudp6\fR).  This string argument is optional.
+
+\fBtcp4\fR, \fBudp4\fR: These arguments are used to bind the service to IPv4
+only.
+.br
+\fBtcp6\fR, \fBudp6\fR: These arguments are used to bind the service to IPv6
+only, if the operating system supports this.
+.br
+\fBtcp\fR, \fBudp\fR: These arguments are used to bind to both IPv4 and IPv6
+if possible.
 .IP "\fBprefork=\fR0" 5
 The number of instances of this service to always have running and
 waiting for a connection (for faster initial response time).  This

